### PR TITLE
Ensure geordiLogger is set before mount

### PR DESCRIPTION
**Pull request by srallen**
Possible bug fix for reported issue here: https://www.zooniverse.org/projects/adrianevans/fossil-finder/talk/49/63272?page=1

This instantiates the Geordi client in `componentWillMount` and updates it on auth change.

Also removed the extraneous `updateUser`

### CODE REVIEW

File: app/partials/app.cjsx [hunk 1]
```diff
@@ -24,9 +24,9 @@ PanoptesApp = React.createClass
     initialLoadComplete: false
     user: null
 
-  updateUser: (user) ->
-    @setState user: user
-
```
Comment (brian-c) [inline app/partials/app.cjsx]: `updateUser` is referenced on lines 14 and 20, too, mind removing those? Then we can close #2740 with this.

Comment (srallen) [inline app/partials/app.cjsx]: Removed

**Comment by brian-c** [comment]
You think it'd be worth removing the `?` from `context.geordi?` (and removing `!!this.context.geordi && ...` checks) in the app so there's no more confusion as to whether or not geordi exists? I'm happy to tack a commit onto this PR if that's alright.

**Comment by srallen** [comment]
Sounds good. Thanks.

**Pull request merged**